src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/3.diff.sql

```diff
@@ -166,17 +163,9 @@ AS
 BEGIN
     SET NOCOUNT ON
 
-    DECLARE @maxSchemaVersion int,
-            @minSchemaVersion int
-
-    Select @maxSchemaVersion = MIN(MaxVersion), @minSchemaVersion = MAX(MinVersion)
+    Select MAX(MinVersion), MIN(MaxVersion)
     FROM dbo.InstanceSchema
     WHERE Timeout > SYSUTCDATETIME()
-
-    SELECT @minSchemaVersion, MAX(Version)
-    FROM dbo.SchemaVersion
-    WHERE Status = 'complete' AND Version <= @maxSchemaVersion
-
```

Comment:
Why do we no longer need to check this?

Reply:
Good question! Tool would need to call this compatibility api to verify if the given version is in compatibility range, if yes, it would proceed to run the sql script for this given version. While status = Current specifies the current version or previous versions for which migration script had already run. I realized while working on the tool side so here is the refactoring.